exporter/exporterhelper/obsexporter.go

```diff
@@ -154,7 +154,7 @@ func (or *ObsReport) StartTracesOp(ctx context.Context) context.Context {
 // EndTracesOp completes the export operation that was started with StartTracesOp.
 func (or *ObsReport) EndTracesOp(ctx context.Context, numSpans int, err error) {
 	numSent, numFailedToSend := toNumItems(numSpans, err)
-	or.recordMetrics(ctx, component.DataTypeTraces, numSent, numFailedToSend)
+	or.recordMetrics(context.Background(), component.DataTypeTraces, numSent, numFailedToSend)
```

Comment:
Is it true that we'd never need the original context inside `recordMetrics`?

Reply:
go 1.21 has a WithoutCancel func I would suggest we move to in the future (whenever 1.22 is out)

Reply:
If we go forward with this can we add a TODO or an issue to track that?

Reply:
Yeah i was planning on filing an issue to track this change

Reply:
added an issue #9049

Reply:
Suggest use of `context.TODO()` here. I understand this solves a short-term problem, but I'd like to ask a bigger question about why we would bypass metric events on canceled contexts. It seems like a recipe for lost time and user confusion, as happened here. If we could convince the Otel-Go SDK to modify this behavior, perhaps restore the `context.TODO()` to the original context.

As you suggested, an alternate SDK with an extension for interpreting context will lose access to baggage, something we could fix in the long run. We have a few internal uses where exporterhelper metrics such as these are being extended from context with header values from the gRPC metadata. (Using [this](https://github.com/lightstep/otel-launcher-go/tree/main/lightstep/sdk/metric#measurementprocessor).)

Reply:
@jmacd i followed @fatsheep9146's suggestion and implemented withoutCancel temporarily in this code